internal/wasm/call_context_test.go

```diff
@@ -147,30 +148,46 @@ func TestCallContext_Close(t *testing.T) {
 	}
 
 	t.Run("calls Context.Close()", func(t *testing.T) {
-		testFS := sysfs.Adapt(testfs.FS{"foo": &testfs.File{}})
-		sysCtx := internalsys.DefaultContext(testFS)
-		fsCtx := sysCtx.FS()
+		for _, concurrent := range []bool{false, true} {
+			concurrent := concurrent
+			t.Run(fmt.Sprintf("concurrent=%v", concurrent), func(t *testing.T) {
+				testFS := sysfs.Adapt(testfs.FS{"foo": &testfs.File{}})
+				sysCtx := internalsys.DefaultContext(testFS)
+				fsCtx := sysCtx.FS()
+
+				_, err := fsCtx.OpenFile(testFS, "/foo", os.O_RDONLY, 0)
+				require.NoError(t, err)
 
-		_, err := fsCtx.OpenFile(testFS, "/foo", os.O_RDONLY, 0)
-		require.NoError(t, err)
+				m, err := s.Instantiate(testCtx, &Module{}, t.Name(), sysCtx)
+				require.NoError(t, err)
 
-		m, err := s.Instantiate(testCtx, &Module{}, t.Name(), sysCtx)
-		require.NoError(t, err)
+				// We use side effects to determine if Close in fact called Context.Close (without repeating sys_test.go).
+				// One side effect of Context.Close is that it clears the openedFiles map. Verify our base case.
+				_, ok := fsCtx.LookupFile(3)
+				require.True(t, ok, "sysCtx.openedFiles was empty")
 
-		// We use side effects to determine if Close in fact called Context.Close (without repeating sys_test.go).
-		// One side effect of Context.Close is that it clears the openedFiles map. Verify our base case.
-		_, ok := fsCtx.LookupFile(3)
-		require.True(t, ok, "sysCtx.openedFiles was empty")
-
-		// Closing should not err.
-		require.NoError(t, m.Close(testCtx))
+				// Closing should not err.
+				if concurrent {
+					hammer.NewHammer(t, 100, 10).Run(func(name string) {
```

Comment:
nit: indentation is getting intense, but ok. if you see a cheap way to reduce this, by extracting a new top-level test, or a function do it.